tests/product/timing_test_decorator.py

```diff
@@ -0,0 +1,65 @@
+import logging
```

Comment:
i'm sorry; i don't see the problem. my idea was to use what you added and build off of that. for example, we run the tests and your change outputs how long the test takes. if it takes a long time, we can use the logger to time some of the functions that test uses and figure out where the problem is.